### ISSUE TITLE
Failed to open Minecraft: EGL: Failed to create context: Arguments are inconsistent

### ISSUE THREAD

**Issue by AlexLexusYTXD** [minecraft-linux/mcpelauncher-manifest#780]
I have Minecraft on macOS 

logs:
**09:22:10 Info  [Launcher] Version: client 07e2986 / manifest 94bc336
09:22:10 Info  [Launcher] CPU: GenuineIntel       Intel(R) Core(TM) i7-2635QM CPU @ 2.00GHz
09:22:10 Info  [Launcher] CPU supports SSSE3: YES
09:22:10 Trace [Launcher] Loading hybris libraries
09:22:10 Trace [REDIRECT] /data/data/com.mojang.minecraftpe to /Users/user/Library/Application Support/mcpelauncher/
09:22:10 Trace [REDIRECT] /data/data/Applications/Minecraft Bedrock Launcher.app/Contents/proc/3200/cmdline to /Users/user/Library/Application Support/mcpelauncher/
09:22:10 Trace [REDIRECT] /data/data/Applications/Minecraft Bedrock Launcher.app/Contents/MacOS/./mcpelauncher-client to /Users/user/Library/Application Support/mcpelauncher/
09:22:10 Trace [REDIRECT] /data/data to /Users/user/Library/Application Support/mcpelauncher/
09:22:10 Trace [LinkerUtils] Loaded OS library libm.dylib
09:22:10 Trace [LinkerUtils] Loaded OS library libz.dylib
09:22:10 Trace [LinkerUtils] Loaded OS library /Applications/Minecraft Bedrock Launcher.app/Contents/Resources/mcpelauncher/lib/native/x86_64/libfmod.dylib
09:22:10 Trace [Launcher] Loading Minecraft library
09:22:11 Error [MinecraftUtils] Failed to load Minecraft: dlopen failed: cannot locate symbol "glBindRenderbuffer" referenced by "/Users/user/Library/Application Support/mcpelauncher/versions/1.19.51.01/lib/x86_64/libminecraftpe.so"...
09:22:11 Info  [Launcher] Loading gamepad mappings
09:22:11 Trace [Launcher] Loading gamepad mappings: /Applications/Minecraft Bedrock Launcher.app/Contents/Resources/mcpelauncher/gamecontrollerdb.txt
09:22:11 Info  [Launcher] Creating window
libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: EGL: Failed to create context: Arguments are inconsistent
Signal 6 received
Backtrace elements: 18
#0 0   mcpelauncher-client                 0x0000000105407418 _ZN12CrashHandler12handleSignalEiPv + 344
#1 1   mcpelauncher-client                 0x0000000105407845 _ZN12CrashHandler15handle_fs_faultEiPvS0_ + 133
#2 2   libsystem_platform.dylib            0x00007fff7c71af5a _sigtramp + 26
#3 3   ???                                 0x000000010dfe1288 0x0 + 4529721992
#4 4   libsystem_c.dylib                   0x00007fff7c4b81ae abort + 127
#5 5   libc++abi.1.dylib                   0x000000010696ffe3 abort_message + 195
#6 6   libc++abi.1.dylib                   0x000000010695112e _ZL28demangling_terminate_handlerv + 238
#7 7   libc++abi.1.dylib                   0x000000010696efa8 _ZSt11__terminatePFvvE + 8
#8 8   libc++abi.1.dylib                   0x00000001069721cb _ZN10__cxxabiv1L12failed_throwEPNS_15__cxa_exceptionE + 27
#9 9   libc++abi.1.dylib                   0x0000000106972145 __cxa_throw + 117
#10 10  mcpelauncher-client                 0x0000000105410f54 _ZN14GLFWGameWindowC2ERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEEii11GraphicsApi + 564
#11 11  mcpelauncher-client                 0x0000000105411a71 _ZN14GLFWGameWindowC1ERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEEii11GraphicsApi + 49
#12 12  mcpelauncher-client                 0x0000000105414aa8 _ZN17GLFWWindowManager12createWindowERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEEii11GraphicsApi + 88
#13 13  mcpelauncher-client                 0x00000001053bfa9a _ZN10FakeLooper16initializeWindowEv + 330
#14 14  mcpelauncher-client                 0x00000001053bf7b5 _ZN10FakeLooper10initWindowEv + 85
#15 15  mcpelauncher-client                 0x00000001050d8ba6 main + 7478
#16 16  libdyld.dylib                       0x00007fff7c40c015 start + 1
#17 17  ???                                 0x0000000000000003 0x0 + 3
Dumping stack...
program failed with unix signal number: 6

Process exited with unexpected exit code: 6**

how to fix this?

**Comment by GameParrot** [comment]
What Mac model and macOS version are you using?

**Comment by ChristopherHX** [comment]
Did this launcher show the loading screen of 1.19.51 before the update?

You can get the version of before the update here: https://github.com/ChristopherHX/osx-packaging-scripts/releases/tag/v0.8.0-394

**Comment by paperlyly** [comment]
I am on Linux, error 6 also happens to me when I leave the game, and at random times

**Comment by RavinMaddHatter** [comment]
I have a similar crash on Steam OS for 1.19.50.02 (X86_64).

SteamOS 20221228.1 Version ID 3.4.4
[1-19-50-02.txt](https://github.com/minecraft-linux/mcpelauncher-manifest/files/10733723/1-19-50-02.txt)


I get a similar crash on 1.19.51, however 1.19.60 seams stable

Crash occurs randomly, Mostly when entering a world. I don't have enough repro's to establish a pattern yet, however with my game pad set up as a game pad i get the crash a lot, with my game pad set up as a keyboard i was able to get into the game place some blocks ETC, then when i switched over to it being a controller (steam deck config switch) when i touched the controls it crashed. 

I need more time messing with things to get a solid case, but it takes time as i cant get the debug dump in big picture, but it is a mess switching between controls in desktop mode. 

As i said i dont have enough data right now to say for sure but i suspect mojang has been changing things in the controller input section and 1.19.50 was minor changes that cause crashed compared to the old version and 1.19.60 moved to a new scheme... they like to do stuff like that....

**Comment by ChristopherHX** [comment]
exit code 6 <=> abort() has been called, the error reported here can never appear on linux.


Anyways I will close this issue if the original author is not willing to provide any requested information within 2 months.

**Comment by RavinMaddHatter** [comment]
Check the crash dump in the text file, exit code 6 on last line

**Comment by ChristopherHX** [comment]
This issue:
```
libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: EGL: Failed to create context: Arguments are inconsistent
Signal 6 received
```
Your problem
```
free(): invalid next size (fast)
Signal 6 received
```
You might no longer see this crash very often in this test build (https://github.com/minecraft-linux/mcpelauncher-client/actions/runs/4168163112)

I hope you get what I mean.

**Comment by RavinMaddHatter** [comment]
Aah. I saw error code 6 in title. I will test tomorrow.  Thanks.

**Comment by AlexLexusYTXD** [comment]
> What Mac model and macOS version are you using?

I am using: 

macOS High Sierra version 10.13.6 

MacBook Pro (15-inch, Early 2011)
Processor 2 GHz Intel Core i7
Memory 16 GB 1333 MHz DDR3
Graphics Intel HD Graphics 3000 512 MB

**Comment by AlexLexusYTXD** [comment]
> Did this launcher show the loading screen of 1.19.51 before the update?

> You can get the version of before the update here: https://github.com/ChristopherHX/osx-packaging-scripts/releases/tag/v0.8.0-394

Yes, before the update my Minecraft 1.19.51 was stuck at 56%.

**Comment by AlexLexusYTXD** [comment]
> This issue:
> 
> ```
> libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: EGL: Failed to create context: Arguments are inconsistent
> Signal 6 received
> ```
> 
> Your problem
> 
> ```
> free(): invalid next size (fast)
> Signal 6 received
> ```
> 
> You might no longer see this crash very often in this test build (https://github.com/minecraft-linux/mcpelauncher-client/actions/runs/4168163112)
> 
> I hope you get what I mean.

I downloaded the test build, although I still got this crash:

15:55:29 Info  [Launcher] Version: client bbb4d76 / manifest 342bd6f
15:55:29 Info  [Launcher] CPU: GenuineIntel       Intel(R) Core(TM) i7-2635QM CPU @ 2.00GHz
15:55:29 Info  [Launcher] CPU supports SSSE3: YES
15:55:29 Trace [Launcher] Loading hybris libraries
15:55:29 Trace [REDIRECT] /data/data/com.mojang.minecraftpe to /Users/user/Library/Application Support/mcpelauncher/
15:55:29 Trace [REDIRECT] /data/data/Applications/Minecraft Bedrock Launcher.app/Contents/proc/24956/cmdline to /Users/user/Library/Application Support/mcpelauncher/
15:55:29 Trace [REDIRECT] /data/data/Applications/Minecraft Bedrock Launcher.app/Contents/MacOS/./mcpelauncher-client to /Users/user/Library/Application Support/mcpelauncher/
15:55:29 Trace [REDIRECT] /data/data to /Users/user/Library/Application Support/mcpelauncher/
15:55:29 Trace [REDIRECT] . to /Users/user/Library/Application Support/mcpelauncher/versions/1.19.51.01/assets/
15:55:29 Trace [LinkerUtils] Loaded OS library libm.dylib
15:55:29 Trace [LinkerUtils] Loaded OS library libz.dylib
15:55:29 Trace [LinkerUtils] Loaded OS library /Applications/Minecraft Bedrock Launcher.app/Contents/Resources/mcpelauncher/lib/native/x86_64/libfmod.dylib
15:55:29 Trace [Launcher] Loading Minecraft library
15:55:30 Error [MinecraftUtils] Failed to load Minecraft: dlopen failed: cannot locate symbol "glBindRenderbuffer" referenced by "/Users/user/Library/Application Support/mcpelauncher/versions/1.19.51.01/lib/x86_64/libminecraftpe.so"...
15:55:30 Info  [Launcher] Loading gamepad mappings
15:55:30 Trace [Launcher] Loading gamepad mappings: /Applications/Minecraft Bedrock Launcher.app/Contents/Resources/mcpelauncher/gamecontrollerdb.txt
15:55:30 Info  [Launcher] Creating window
libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: EGL: Failed to create context: Arguments are inconsistent
Signal 6 received
Backtrace elements: 18
#0 0   mcpelauncher-client                 0x0000000107c03d08 _ZN12CrashHandler12handleSignalEiPv + 344
#1 1   mcpelauncher-client                 0x0000000107c04135 _ZN12CrashHandler15handle_fs_faultEiPvS0_ + 133
#2 2   libsystem_platform.dylib            0x00007fff5db1ef5a _sigtramp + 26
#3 3   ???                                 0x000000010d13d314 0x0 + 4514370324
#4 4   libsystem_c.dylib                   0x00007fff5d8bc1ae abort + 127
#5 5   libc++abi.1.dylib                   0x00000001091a5ff3 abort_message + 195
#6 6   libc++abi.1.dylib                   0x000000010918713e _ZL28demangling_terminate_handlerv + 238
#7 7   libc++abi.1.dylib                   0x00000001091a4fb8 _ZSt11__terminatePFvvE + 8
#8 8   libc++abi.1.dylib                   0x00000001091a81db _ZN10__cxxabiv1L12failed_throwEPNS_15__cxa_exceptionE + 27
#9 9   libc++abi.1.dylib                   0x00000001091a8155 __cxa_throw + 117
#10 10  mcpelauncher-client                 0x0000000107c0d844 _ZN14GLFWGameWindowC2ERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEEii11GraphicsApi + 564
#11 11  mcpelauncher-client                 0x0000000107c0e361 _ZN14GLFWGameWindowC1ERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEEii11GraphicsApi + 49
#12 12  mcpelauncher-client                 0x0000000107c11398 _ZN17GLFWWindowManager12createWindowERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEEii11GraphicsApi + 88
#13 13  mcpelauncher-client                 0x0000000107bbc59a _ZN10FakeLooper16initializeWindowEv + 330
#14 14  mcpelauncher-client                 0x0000000107bbc2b5 _ZN10FakeLooper10initWindowEv + 85
#15 15  mcpelauncher-client                 0x00000001078c6420 main + 7936
#16 16  libdyld.dylib                       0x00007fff5d810015 start + 1
#17 17  ???                                 0x0000000000000003 0x0 + 3
Dumping stack...
program failed with unix signal number: 6

Process exited with unexpected exit code: 6

**Comment by ChristopherHX** [comment]
Thank you for providing the requested information.

I'm going to revert a change..., in the next update.

Seems like your intelhd3000 doesn't support emulating opengl es3. My intelhd4000 had no problem.

**Comment by AlexLexusYTXD** [comment]
> Thank you for providing the requested information.
> 
> I'm going to revert a change..., in the next update.
> 
> Seems like your intelhd3000 doesn't support emulating opengl es3. My intelhd4000 had no problem.

Yay!

**Comment by AlexLexusYTXD** [comment]
> Thank you for providing the requested information.
> 
> I'm going to revert a change..., in the next update.
> 
> Seems like your intelhd3000 doesn't support emulating opengl es3. My intelhd4000 had no problem.

Can I download an older version of MC Bedrock launcher so i could play while you're working on the fixed version?

**Comment by ChristopherHX** [comment]
> Can I download an older version of MC Bedrock launcher so i could play while you're working on the fixed version?

Just use that version and select 1.18.12 or lower in the profile editor.

> > Did this launcher show the loading screen of 1.19.51 before the update?
> 
> > You can get the version of before the update here: https://github.com/ChristopherHX/osx-packaging-scripts/releases/tag/v0.8.0-394
> 
> Yes, before the update my Minecraft 1.19.51 was stuck at 56%.

**Comment by AlexLexusYTXD** [comment]
Still crashing :v

**Comment by ChristopherHX** [comment]
And you are using https://github.com/ChristopherHX/osx-packaging-scripts/releases/tag/v0.8.0-394???

I mean you said it doesn't crash, so 1.18.12 should open without the 56% issue

**Comment by AlexLexusYTXD** [comment]
> And you are using https://github.com/ChristopherHX/osx-packaging-scripts/releases/tag/v0.8.0-394???
> 
> I mean you said it doesn't crash, so 1.18.12 should open without the 56% issue
Nvm, I clicked on https://github.com/ChristopherHX/osx-packaging-scripts/releases/tag/v0.8.0-394??? and downloaded the launcher, now it works! :D

**Comment by ChristopherHX** [comment]
@AlexLexusYTXD Does https://github.com/minecraft-linux/game-window/suites/11050967422/artifacts/561708244 fix your problem?

_You know I cannot reproduce your error on my hardware_

**Comment by AlexLexusYTXD** [comment]
> @AlexLexusYTXD Does https://github.com/minecraft-linux/game-window/suites/11050967422/artifacts/561708244 fix your problem?
> 
> _You know I cannot reproduce your error on my hardware_

Thank you, it worked!

**Comment by ChristopherHX** [comment]
Fixed in 0.8.5
